### ISSUE TITLE
[windows] SSHFS does not deny access to folders that a user does not have permissions for

### ISSUE THREAD

**Issue by sharder996** [canonical/multipass#2926]
**Describe the bug**
When creating an SMB mount, Multipass will check to see if the user has permissions for that folder and only create the mount if the user does. On the other hand, SSHFS mounts does not check and will create mounts to restricted directories.

**To Reproduce**
1. `multipass mount -t native C:\Windows\CSC\ foo:CSC_native`
2. `multipass mount C:\Windows\CSC\ foo:CSC`

**Expected behavior**
Both mounts should have the same behavior

**Logs**
Please provide logs from the daemon, see [accessing logs](https://multipass.run/docs/accessing-logs) on where to find them on your platform.

**Additional info**
 - OS: `Windows`
- `multipass version`: `1.11`
- `multipass get local.driver`: `hyper-v`

**Comment by conioh** [comment]
That's a "feature".  See: https://multipass.run/docs/privileged-mounts

**Comment by tobe2098** [comment]
This is still relevant as long as the behavior is unchanged. Is it still the case @sharder996 ? 
Updated docs link for previous comment: https://documentation.ubuntu.com/multipass/stable/explanation/mount/

**Comment by ricab** [comment]
Yeah, I think it's still a problem. I don't think ID mappings even have an effect on Windows...
